osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmap.cs

```diff
@@ -57,8 +60,26 @@ private void load(SongSelect songSelect, BeatmapManager manager, BeatmapSetOverl
                 editRequested = songSelect.Edit;
             }
 
-            if (manager != null)
-                hideRequested = manager.Hide;
+            if (beatmapManager != null)
+                hideRequested = beatmapManager.Hide;
+
+            var starCounterChildren = new List<Drawable>();
```

Comment:
I'd add `starCounter` lower down alongside the entire hierarchy using initializers and only do `Insert(0, ...)` on the `DrawableRank` conditionally if there is a rank to display. Just seems to be more consistent with how things are done elsewhere.